src/Explorer/Notebook/NotebookContainerClient.ts

```diff
@@ -84,6 +93,13 @@ export class NotebookContainerClient {
           "Connection lost with Notebook server. Attempting to reconnect..."
         );
       }
+      if (userContext.features.notebooksTemporarilyDown === false && userContext.features.phoenix === true) {
+        const connectionStatus: ContainerConnectionInfo = {
+          status: ConnectionStatusType.Failed,
+        };
+        useNotebook.getState().resetConatinerConnection(connectionStatus);
+        useNotebook.getState().setIsRefreshed(true);
```

Comment:
>setIsRefreshed [](http://example.com/codeflow?start=31&length=14)

whats this?

Reply:
When connection to container fails we are changing the status to reconnect also need to refresh the notebook tree[used this flag for it]. i.e., to remove temporary notebooks. Otherwise will end up in errors in each execution.